### DISCUSSION TITLE
Welcome to arcfile discussions!

### DISCUSSION THREAD

**Discussion by jrpowers** [arcturus-industries/arcfile#1] [General]
# 👋 Introducing ARC

### ARC file is a side project at [Arcturus Industries](https://arcturus.industries) where we aim to create an open file format for 3D vision research and production.

At Arcturus Industries, we work on things like visual-inertial SLAM (Simultaneous Localization and Mapping), depth estimation, and scene reconstruction. We want ARC file to work well for those tasks, but also a wider range of tasks like data capture for deep learning research topics like neural rendering, and view synthesis. And likely will be suitable to other tasks like volumetric video, and more.

ARC will be open to everyone and permissively licensed.

We're just getting started, and plan to develop ARC publicly. As of this writing, it's nothing more than an empty repository, but we do have some guiding principles ...

### Principles behind ARC:
* ARC should not just be a format, but also a tool suite to make it easy to inspect and manipulate ARC format
* ARC should be interoperable with other popular formats
* ARC should be extremely efficient both size-wise and overhead-wise
* ARC should be easy to integrate into other projects, and require minimal dependencies
* ARC should cross-platform, yet should leverage native hardware acceleration on Mac, Linux, Windows, iOS, and Android

### Why not ROSbag, ASL, Protobuf, ...

In our experience working in the field creating computer vision products like RedLaser, Structure Sensor, Canvas, and Skanect, we haven't encountered a format that we felt fits our needs. They are either too cumbersome to integrate, don't make it seamless to leverage hardware acceleration, and/or are space inefficient. If we've overlooked a great option that already exists, let us know!

Things like Protobuf, Flatbuffers, CapnProto are serialization layers, and ARC may well use one of these, but ARC aims to be a higher-level container and toolset on top of pure serialization.

If this project interests you, please introduce yourself! We'd love to hear your ideas and questions. And let us know if you'd like to get involved!

With that, let's get started!

![Hacking](https://i.giphy.com/media/ekjmhJUGHJm7FC4Juo/giphy.webp)

**Comment by ironandsilk** [comment]
This sounds useful. I see a need for common file format between capture types. Especially as the hardware often supports different distances and resolutions.

**Reply by jrpowers** [reply]
Thanks!

Would love to hear which types of capture hardware you'd like to see the format optimized to handle. I think the underlying serialization should be as "raw" as possible given the hardware, and then have fast realtime accessors that convert to unified representations (like raw disparity --> float depth for a depth camera)

**Reply by ironandsilk** [reply]
Naturally for LiDAR consumer and home use-cases around 2meters.
The captures created for enterprise with tools like Matterport Pro, NavVis, Faro sound useful (PTC shares more in this [AR NY meet up vid at 1:17:43](https://www.linkedin.com/feed/update/urn:li:activity:6744717511289307136/) ),
but enterprises should already have a "master plan" of their stadium or factory.

Some way to match and solve for Android fragmentation and connect spaces between Android <> iOS can be valuable. Like Unity says "build once and deploy everywhere." We're going to have trouble in defining "where" as a location in space and not just OS.

**Reply by pablospe** [reply]
Great initiative! 

Is there a roadmap for this project? Just wondering when it will be some initial format proposals.

I also agree with Faro, NavVis, Leica should be supported. I think all of them support e57 files, which might be an interesting intermediate format for these types of devices.

**Reply by pablospe** [reply]
What are your comments about [Kapture format](https://github.com/naver/kapture/blob/main/kapture_format.adoc)? 

It seems having the same goals.

**Reply by jrpowers** [reply]
Thanks for pointing out Kapture format, we hadn't seen this yet! Kapture looks like it shares a lot of spirit with ARC, but I see a few things we'd want to do differently:
* Pack everything into a single flat file for easy transport
* Support hardware-optimized encode & decode, including video formats (h265, especially)
* Support embedding serialization & deserialization into native applications running C++ (likely alongside Python wrappers)

**Reply by jrpowers** [reply]
The closest thing to a roadmap right now is this list of Format Requirements under consideration -- please feel free to contribute comments!

https://github.com/arcturus-industries/arcfile/projects/1

**Reply by pablospe** [reply]
Any news about this format?
